backend/inspirehep/sitemap/cli.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+import os
+from itertools import islice
+
+import click
+from flask import current_app, render_template, url_for
+from flask.cli import with_appcontext
+
+from inspirehep.utils import get_inspirehep_url
+
+from .sitemap import generate_sitemap_items
+
+
+@click.group()
+def sitemap():
+    """Command group to handle file operations."""
+
+
+def get_sitemap_page_filename(page):
+    return f"sitemap{page}.xml"
+
+
+def get_sitemap_page_absolute_url(page):
+    base_url = get_inspirehep_url()
+    return f"{base_url}/{get_sitemap_page_filename(page)}"
+
+
+def write_sitemap_page_content(page, page_content):
+    path = current_app.config["SITEMAP_FILES_PATH"]  # FIXME: should we mkdir -p?
+    file_path = os.path.join(path, get_sitemap_page_filename(page))
+    with open(file_path, "w+") as file:
+        file.write(page_content)
+
+
+@sitemap.command(
+    help="Generates sitemaps for records that should be indexed by search engines"
+)
+@with_appcontext
+def generate():
+    page_size = current_app.config["SITEMAP_PAGE_SIZE"]
+
+    sitemap_items = iter(generate_sitemap_items())
```

Comment:
Why you are creating iterator from generator? At the end it gives you the same thing.

Reply:
some lines in this function are adapted from zenodo https://github.com/zenodo/zenodo/blob/44ee8568b0641eea96bdefbb85d50643564df9b2/zenodo/modules/sitemap/tasks.py#L36.

The PR is WIP at the moment, my plan is to write tests first and then make this kind of changes.

Reply:
:+1: